kolena/dataset/evaluation.py

```diff
@@ -169,8 +169,11 @@ def upload_results(
     :param dataset: The name of the dataset.
     :param model: The name of the model.
     :param results: Either a DataFrame or a list of tuples, where each tuple consists of
-                    an eval configuration and a DataFrame.
-    :return: None
+                    an eval configuration and a DataFrame. The DataFrame type has been aliased to be either a pandas
+                    DataFrame or an iterator of pandas DataFrame, this allows for batch upload when passing iterators.
+
+    :return: An UploadResultsResponse object containing information about the number of result rows created
+             and updated
```

Comment:
It's not very meaningful to refer to `UploadResultsResponse` when we don't have the type documented -- how will users know what the fields to look into actually are?

Reply:
I will add the doc for it

Reply:
This actually raises a larger problem: `UploadResultsResponse` is an internal API object and shouldn't leak into the external-facing API as the response from `upload_results`.

What are users expected to do with an `UploadResultsResponse`? We should consider changing this function to return `None`.

Reply:
@gordonhart What would you recommend if we want to make `UploadResultsResponse` public?

Reply:
Or can we create a public data model, I still think number of rows created/ updated is valuable info